ouroboros-consensus/test-consensus/Test/Consensus/Mempool.hs

```diff
@@ -831,9 +834,38 @@ prop_TxSeq_lookupByTicketNo_sound smalls small =
     txseq =
         foldl' (TxSeq.:>) TxSeq.Empty $ map mkTicket haystack
 
-    mkTicket x = TxTicket x (mkTicketNo x)
+    mkTicket x = TxTicket x (mkTicketNo x) 0
     mkTicketNo = TicketNo . toEnum
 
+-- | Test that the 'fst' of the result of 'splitAfterTxSize' only contains
+-- 'TxTicket's whose summed up transaction sizes are less than or equal to
+-- that of the 'TxSizeInBytes' which the 'TxSeq' was split on.
+prop_TxSeq_splitAfterTxSize :: [Positive TxSizeInBytes] -> Property
```

Comment:
Alternatively or *additionally*, write a `splitAfterTxSizeSpec`, which is just a simple, naive implementation (e.g., https://github.com/input-output-hk/ouroboros-network/pull/1342#discussion_r357014860), and check whether it produces the same outputs as `splitAfterTxSize` for the same inputs.